### ISSUE TITLE
Azure functions should support versions other than [~1 beta]

### ISSUE THREAD

**Issue by ruhkopf** [terraform-providers/terraform-provider-azurerm#1731]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.azurerm v1.12.0

### Affected Resource(s)

* azurerm_function_app

### Expected Behavior
I should be able to enter a specific version such as `2.0.11933-alpha`. Currently I have to do this manually in the Azure portal. We currently can't use `beta`, because that version is broken for NodeJS deployments at the moment.

### Actual Behavior
The following error is shown: ` expected version to be one of [~1 beta], got 2.0.11933-alpha`.

I suggest to either remove the validation or adjust it so that you can specify values other than `~1` and `beta`.

What's frustrating about this bug is that when I change the version on Azure and I run `terraform apply` whith `beta` as the version, TF wants to revert:

```
  ~ azurerm_function_app.my-functions-app
      version:                                   "~2" => "beta"
```
So I have to manually go in and change it back again each time I deploy changes. Is there a workaround to disable validations until it is fixed?

**Comment by JimPaine** [comment]
I hit the same issue when trying to pin my app for this [breaking change](https://github.com/Azure/app-service-announcements/issues/112)

I have submitted a pull request to remove the validation allowing for furture versions and pinning of versions to work. 

After looking through the Azure REST API I could see the validation was specific to this terraform provider.

[Pull Request](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1872)

**Comment by tombuildsstuff** [comment]
Closing this since this has been fixed via #1872 which shipped as a part of v1.14
